src/Features/CSharp/Portable/MoveToResx/CSharpMoveToResxService.cs

```diff
@@ -0,0 +1,459 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp.MoveToResx;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+internal sealed class CSharpMoveToResxDiagnosticAnalyzer : DiagnosticAnalyzer
```

Comment:
i'm surpirsed this isn't subclassing one of our common DiagAnalyzer types (which take cares of all the thigns like title/messageformat/descriptor/etc.